Realm/RLMObjectStore.mm

```diff
@@ -162,6 +162,27 @@ void RLMRealmSetSchema(RLMRealm *realm, RLMSchema *targetSchema, bool verify) {
     }
 }
 
+// try to get all of the tables for the realm
+// if they all exist, sets the realm's schema to targetSchema and returns true
+// returns false if any are missing and does not change the schema
+static bool RLMRealmGetTables(RLMRealm *realm, RLMSchema *targetSchema) {
+    if (!RLMRealmHasMetadataTables(realm)) {
+        return false;
+    }
+
+    for (RLMObjectSchema *objectSchema in targetSchema.objectSchema) {
+        NSString *tableName = RLMTableNameForClass(objectSchema.className);
+        TableRef table = realm.group->get_table(tableName.UTF8String);
+        if (!table) {
+            return false;
+        }
+        objectSchema.table = table.get();
+    }
+
+    RLMRealmSetSchema(realm, targetSchema, true);
```

Comment:
Think it might be better if RLMSetSchema is called from the calling function instead of here. Same probably goes with RLMCreateTables. The concern is that it isn't clear that RLMRealmSetSchema is being called based on these function names. Could also rename both methods to RLMRealmTryGetTablesAndSetSchema and RLMCreateTablesAndSetSchema but those sound horrible